custom_dataset.py

```diff
@@ -0,0 +1,408 @@
+        """
+        # Get label and image Tensor
+        label = self._get_label(file_path)
+        img = tf.io.read_file(file_path)
+        img = self._decode_img(img)
+        return img, label
+
+    def _process_img_path(self, file_path):
+        """Similar to self._process_img_path. Do not require label path! Uses
+        first section of file name as label.
+
+        Args:
+            file_path: Str, path to image file.
+
+        Returns:
+            img: tf.Tensor: image tensor in type tf.float32.
+            label: tf.Tensor: label tensor represented by file name.
+        """
+        label = tf.strings.split(file_path, os.path.sep)[-1]
+        img = tf.io.read_file(file_path)
+        img = self._decode_img(img)
+        return img, label
+
+    def _convert_format(self, img, label):
+        """Convert image format from RGB to grayscale or grayscale to RGB.
+
+        Args:
+            img: tf.Tensor: image tensor in type tf.float32.
+            label: tf.Tensor: label tensor.
+
+        Returns:
+            img: tf.Tensor: image tensor in type tf.float32.
+            label: tf.Tensor: label tensor.
+        """
+        # Convert between graycale and rgb
+        if self._GRAYSCALE_IN and not self._GRAYSCALE_OUT:
+            # If input is grayscale and output is rgb
+            img = tf.image.grayscale_to_rgb(img)  # use tensorflow function
+        elif not self._GRAYSCALE_IN and self._GRAYSCALE_OUT:
+            # If input is rgb and output is grayscale
+            # img = tf.reduce_mean(img, axis=2)
+            img = tf.image.rgb_to_grayscale(img)
+        return img,label
+
+    def _augment(self, img, label):
+        """Data augmentation. Two augmentation method are carried out:
+            1. Random rotate: randomly rotate image by x degree. x follows
+                normal distribution with 0 as mean as described in config file.
+            2. Random zoom: crop (zoom in) on the image by x percent. x follows
+                normal distribution with 0 as mean as described in config file.
+
+        Args:
+            img: tf.Tensor: image tensor in type tf.float32.
+            label: tf.Tensor: label tensor.
+
+        Returns:
+            img: tf.Tensor: image tensor in type tf.float32.
+            label: tf.Tensor: label tensor.
+        """
+        # Randomly rotate by -3 and 3 degrees
+        if self._RANDOM_ROTATE:
+            # Follows normal distribution whose magnitude is more than 2 standard
+            # deviations from the mean are dropped and re-picked
+            degree = tf.random.truncated_normal(shape=[],
+                                                stddev=self._ROTATE_STDDEV)
+            # Rounds half to even. Also known as bankers rounding.
+            degree = tf.math.round(degree)
+            img = tfa.image.rotate(img, degree)
+
+        # Randomly zoom in on image
+        if self._RANDOM_ZOOM:
+            # Generate 5 crop settings, ranging from a 0% to n% crop.
```

Comment:
could you wrap the code inside the if block as a separate utility function which take zoom_percent, height, width, img, and zoom_stddev as parameter and call from here? Make sure that utility function does not depend on any private data from self.